src/dom_types.rs

```diff
@@ -766,63 +770,129 @@ impl<Ms: Clone> Node<Ms> {
     pub fn add_class(&mut self, name: impl Into<Cow<'static, str>>) -> &mut Self {
-        if let Node::Element(el) = self {
-            el.add_class(name);
+        if let Node::Element(e) = self {
+            e.add_class(name);
         }
         self
     }
 
     /// See `El::add_style`
     pub fn add_style(&mut self, key: impl Into<St>, val: impl Into<CSSValue>) -> &mut Self {
-        if let Node::Element(el) = self {
-            el.add_style(key, val);
+        if let Node::Element(e) = self {
+            e.add_style(key, val);
         }
         self
     }
 
     /// See `El::add_listener`
     pub fn add_listener(&mut self, listener: Listener<Ms>) -> &mut Self {
-        if let Node::Element(el) = self {
-            el.add_listener(listener);
+        if let Node::Element(e) = self {
+            e.add_listener(listener);
         }
         self
     }
 
     /// See `El::add_text`
     pub fn add_text(&mut self, text: impl Into<Cow<'static, str>>) -> &mut Self {
-        if let Node::Element(el) = self {
-            el.add_text(text);
+        if let Node::Element(e) = self {
+            e.add_text(text);
         }
         self
     }
 
     /// See `El::replace_text`
     pub fn replace_text(&mut self, text: impl Into<Cow<'static, str>>) -> &mut Self {
-        if let Node::Element(el) = self {
-            el.replace_text(text);
+        if let Node::Element(e) = self {
+            e.replace_text(text);
         }
         self
     }
 
     /// See `El::get_text`
     pub fn get_text(&self) -> String {
         match self {
-            Node::Element(el) => el.get_text(),
-            Node::Text(text) => text.text.to_string(),
-            _ => "".to_string(),
+            Node::Element(e) => e.get_text(),
+            Node::Text(t) => t.get_text(),
+            _ => "".to_owned(),
         }
     }
 }
+// Workspace manipulations.
+impl<Ms> Node<Ms> {
+    pub(crate) fn strip_own_and_children_ws_nodes(&mut self) -> &mut Self {
+        match self {
+            Node::Element(e) => e.strip_own_and_children_ws_nodes(),
+            Node::Text(t) => t.strip_ws_node(),
+            _ => (),
+        }
+        self
+    }
+}
+// Convenience methods for `Node` a la `Result` and `Option`.
+impl<Ms> Node<Ms> {
+    pub fn new_text(text: impl Into<Cow<'static, str>>) -> Self {
+        Node::Text(Text::new(text))
+    }
 
-impl<Ms: 'static + Clone, OtherMs: 'static + Clone> MessageMapper<Ms, OtherMs> for Node<Ms> {
+    pub fn is_text(&self) -> bool {
+        if let Node::Text(_) = self {
+            true
+        } else {
+            false
+        }
+    }
+    pub fn is_el(&self) -> bool {
+        if let Node::Element(_) = self {
+            true
+        } else {
+            false
+        }
+    }
+    pub fn is_empty(&self) -> bool {
+        if let Node::Empty = self {
+            true
+        } else {
+            false
+        }
+    }
+
+    pub fn el(self) -> Result<El<Ms>, Option<Text>> {
```

Comment:
Return value seems a little bit weird to me, I would expect something like:
```rust
pub fn el(self) -> Option<El<Ms> { .. }
```
or
```rust
pub fn el(self) -> Result<El<Ms>, String> { .. }
pub fn el(self) -> Result<El<Ms>, ()> { .. }
```

---

The same comment for methods `text` and `empty`.

Reply:
Changed in a different PR (#271).